ui/src/app/shared/components/dropdown/dropdown.tsx

```diff
@@ -19,112 +20,107 @@ export interface DropDownState {
+                    setTop(newState.top);
                 }
             })
         ];
-    }
 
-    public componentWillUnmount() {
-        (this.subscriptions || []).forEach(s => s.unsubscribe());
-        this.subscriptions = [];
-    }
+        return () => {
+            (subscriptionsRef.current || []).forEach(s => s.unsubscribe());
+            subscriptionsRef.current = [];
+        };
+    }, []);
 
-    public close() {
-        this.setState({opened: false});
-    }
-
-    private refreshState() {
-        const anchor = this.el.querySelector('.argo-dropdown__anchor') as HTMLElement;
-        const {top, left} = anchor.getBoundingClientRect();
+    const refreshState = () => {
+        const content = contentRef.current;
+        const anchor = anchorRef.current;
         const anchorHeight = anchor.offsetHeight + 2;
+        const {top: anchorTop, left: anchorLeft} = anchor.getBoundingClientRect();
+        const newState = {top: anchorTop, left: anchorLeft, opened};
 
-        const newState = {left: this.state.left, top: this.state.top, opened: this.state.opened};
         // Set top position
-        if (this.content.offsetHeight + top + anchorHeight > window.innerHeight) {
-            newState.top = top - this.content.offsetHeight - 2;
+        if (content.offsetHeight + anchorTop + anchorHeight > window.innerHeight) {
+            newState.top = anchorTop - content.offsetHeight - 2;
         } else {
-            newState.top = top + anchorHeight;
+            newState.top = anchorTop + anchorHeight;
         }
 
         // Set left position
-        if (this.content.offsetWidth + left > window.innerWidth) {
-            newState.left = left - this.content.offsetWidth + anchor.offsetWidth;
+        if (content.offsetWidth + anchorLeft > window.innerWidth) {
+            newState.left = anchorLeft - content.offsetWidth + anchor.offsetWidth;
         } else {
-            newState.left = left;
+            newState.left = anchorLeft;
         }
+
         return newState;
-    }
+    };
 
-    private open() {
-        if (!this.content || !this.el) {
+    const open = () => {
+        if (!contentRef.current || !anchorRef.current) {
             return;
         }
 
-        const newState = this.refreshState();
+        const newState = refreshState();
+
         newState.opened = true;
-        this.setState(newState);
-        dropDownOpened.next(this);
-    }
+        setOpened(newState.opened);
+        setLeft(newState.left);
+        setTop(newState.top);
+        dropDownOpened.next(DropDown);
+    };
+
+    const close = () => {
+        setOpened(false);
+    };
+
+    return (
+        <div className='argo-dropdown' ref={anchorRef}>
+            <div
+                qe-id={qeId}
+                className='argo-dropdown__anchor'
+                onClick={event => {
+                    open();
+                    event.stopPropagation();
+                }}>
+                <Anchnor />
```

Comment:
```suggestion
                <anchor />
```

if that doesn't work I think you can use the same syntax as `children`:

```suggestion
                {anchor}
```

Reply:
```ts
export interface DropDownProps {
    isMenu?: boolean;
    anchor: () => JSX.Element;
    children: ReactNode;
    qeId?: string;
}
```
```suggestion
                {anchor()}
```
When I tested, they all returned errors, but when I changed the 'anchor' type of 'DropDownProps' to () => JSX.Element and added {anchor()}, it worked 👍 Is it ok to proceed like this?

Reply:
Yea as far as I can tell, the parent's usage of `anchor` is indeed a straight element.

Since you changed this already, could you go a step further and just make it a plain `JSX.Element` with no function? Would have to change the parent as well. It would be a good simplification for both to remove an unnecessary function!

Reply:
- Thanks for the great idea! I'll remove the function and post it right away 👍